src/foam/core/CSV.js

```diff
@@ -0,0 +1,359 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.csv',
+  name: 'Outputter',
+
+  documentation: 'CSV Outputter.',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'buf_',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'includeQuotes',
+      help: 'If true, values are quoted with the `"` char',
+      value: true
+    },
+    {
+      class: 'String',
+      name: 'delimiter',
+      value: ','
+    },
+    {
+      class: 'String',
+      name: 'nestedObjectSeperator',
+      value: '__'
+    },
+    {
+      class: 'StringArray',
+      name: 'nestedObjectNames'
```

Comment:
But why do you need to store it? This will cause conflicts if you reuse the CSV outputter to output more than one class. So if someone used foam.csv.Compact, which is persistent, then the headers would be corrupted.